ui/app/build-configs/detail/details-tab/pnc-build-config-details-editor.html

```diff
@@ -38,6 +38,32 @@
             <span class="help-block" ng-show="editBuildConfigForm.scmRevision.$invalid && editBuildConfigForm.scmRevision.$touched">Required field.</span>
           </div>
 
+          <!-- Validation: This field is optional-->
+          <div class="form-group"
```

Comment:
It would be much better if the 2 step product version selection code was encapsulated in its own reusable component:
- It is used in multiple places and it seems like a lot of code is copy and pasted to acheive this. The parent component then gets poluted with logic like the "checkVersionData" function which should be encapsulated in the new component.
- It also makes the codebase harder to maintain as fixes may have to take place in multiple locations, this often means they are only fixed at the site reported in a bug report by whoever picks up the ticket!

Reply:
Hi Alex(@alexcreasy ),

Thanks for the review!

I tried to encapsulate it in an individual module, there are 3 usages for the product versions but they are somehow different: 

1. **New Build Config wizard**(https://github.com/DnsZhou/pnc/blob/NCL-5751-P2/ui/app/build-configs/directives/pnc-create-build-config-product-version-form/pnc-create-build-config-product-version-form.html#L21-L33), 

2. **Create Group Config page**(https://github.com/DnsZhou/pnc/blob/NCL-5751-P2/ui/app/group-configs/create/pnc-group-config-create-page.html#L38-L54) 

3. **Build Config detail tab**(which is the code you mentioned here)

The biggest difference is that the UI layout it use was different, e.g., 2. **Create Group Config page** uses "col-sm-3 col-lg-2" to control the label and for the input field it is "col-sm-9 col-lg-7". But 1. **New Build Config wizard** do not have such layouts.

So I can write a new component for 1. **New Build Config wizard** and 3. **Build Config detail tab**, and will launch another review request to you after I finished and commit that.

Cheers!

-Dennis

Reply:
Hi @alexcreasy , there is already existing jira (NCL-5875) for this and I think it's already in progress and @DnsZhou is working on it

Reply:
Hi @DnsZhou if layout customisation is the only problem, I think we should be able to solve it using some custom parameter at least, let's move some additional investigation into jira

Reply:
Thanks @matedo1 . 

So @alexcreasy if you don't mind, I think we can proceed the encapsulation of the product version component in jira NCL-5875, for this PR I will just fix the second issue you mentioned and merge the code after you and Martin reviewed the fix, so that we can unblock Dominik's task about product version link.

Thanks!
-Dennis